features/0794-did-rotate/README.md

```diff
@@ -0,0 +1,112 @@
+# Aries RFC 0794: DID Rotate 1.0
+
+- Authors: [Sam Curren](telegramsam@gmail.com)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2023-08-29 
+- Status Note: 
+- Start Date: 2023-08-18 
+- Tags: [feature](/tags.md#feature), [protocol](/tags.md#protocol)
+- URI: https://didcomm.org/did-rotate/1.0
+
+## Summary
+
+This protocol signals the change of DID in use between parties.
+
+This protocol is only applicable to DIDComm v1 - in DIDComm v2 use the more efficient DID Rotation header.
+
+## Motivation
+
+This mechanism allows a party in a relationship to change the DID in use for the relationship. This may be used to switch DID methods,
+but also to switch to a new DID within the same DID method. Inspired by (but different from) the DID rotation feature of the DIDComm Messaging (DIDComm v2) spec.
+
+## Tutorial
+
+### Name and Version
+
+DID Rotate 1.0
+
+URI: https://didcomm.org/did-rotate/1.0/<messageType>
+
+### Roles
+
+**rotating_party**: this party is rotating the DID in use for this relationship. They send the `rotate` message.
+
+**observing_party**: this party is notified of the DID rotation
+
+### Messages
+
+#### Rotate 
+
+Message Type URI: https://didcomm.org/did-rotate/1.0/rotate
+
+`to_did`: The new DID to be used to identify the **rotating_party**
+
+```json
+{
+    "@id": "123456780",
+    "@type": "https://didcomm.org/did-rotate/1.0/rotate",
+    "to_did": "did:example:newdid",
+
+}
+```
+
+The **rotating_party** is expected to receive messages on both the existing and new DIDs and their associated keys for a reasonable period that MUST extend at least until the following `ack` message has been received
+
+This message MUST be sent using authcrypt or as a signed message in order to establish the provenance of the new DID. Proper provenance prevents injection attacks that seek to take over a relationship. Any rotate message received without being authcrypted or signed MUST be discarded and not processed.
+
+DIDComm v1 uses public keys as the outer message identifiers. This means that rotation to a new DID using the same public key will not result in a change for new inbound messages. The **observing_party** must not assume that the new DID uses the same keys as the existing relationship.
+
+#### Ack
+
+Message Type URI: https://didcomm.org/did-rotate/1.0/ack
+
+This message is still sent to the prior DID to acknowledge the receipt of the rotation. Following messages will be sent to the new DID.
+
+`to_did`: The new DID to be used to identify the **rotating_party**
+
+```json
+{
+    "@id": "123456780",
+    "@type": "https://didcomm.org/did-rotate/1.0/ack",
+    "to_did": "did:example:newdid",
```

Comment:
Is it needed to repeat the `to_did` field to confirm the reception of it, or should this message follow the convention including status and thread id? For example:

```
"status": "OK",
  "~thread": {
    "thid": "rotate-did-message-id",
  }
```

Reply:
Good topic for discussion in the Aries WG call. I included it as a bit of redundant info, but maybe it shouldn't be.

Reply:
fixed in next commit